src/sidebar/store/modules/annotations.js

```diff
@@ -7,20 +7,30 @@
  * @typedef {import('../../../types/api').Annotation} Annotation
  */
 
+/**
+ * @typedef AnnotationStub
+ * @prop {string} [id] - service-provided identifier if annotation has been
+ *       persisted to the service
+ * @prop {string} [$tag] - local-generated identifier
+ */
+
 import { createSelector } from 'reselect';
 
 import * as metadata from '../../util/annotation-metadata';
-import * as arrayUtil from '../../util/array';
+import { countIf, toTrueMap, trueKeys } from '../../util/collections';
 import * as util from '../util';
 
 import route from './route';
 
 /**
  * Return a copy of `current` with all matching annotations in `annotations`
- * removed.
+ * removed (matched on identifier—`id` or `$tag`)
  *
  * Annotations in `annotations` may be complete annotations or "stubs" with only
  * the `id` field set.
+ *
+ * @param {Annotation[]} current
+ * @param {(Annotation|AnnotationStub)[]} annotations
```

Comment:
Since `AnnotationStub` is a subset of `Annotation` you can simplify this to `@param {AnnotationStub[]} annotations`. This is an example of the way types are compatible as long as the "shapes" match up.